packages/gatsby-source-nacelle/gatsby-node.js

```diff
@@ -1,18 +1,18 @@
 const sourceNodes = require('./src/source-nodes');
 const typeDefs = require('./src/type-defs');
 const { createRemoteImageFileNode, cmsPreviewEnabled } = require('./src/utils');
-const { nacelleClient } = require('./src/services');
+const { nacelleClient: createNacelleClient } = require('./src/services');
 
 exports.pluginOptionsSchema = ({ Joi }) => {
   return Joi.object({
-    nacelleSpaceId: Joi.string()
-      .required()
-      .description('Space ID from the Nacelle Dashboard'),
-    nacelleGraphqlToken: Joi.string()
-      .required()
-      .description('GraphQL Token from the Nacelle Dashboard'),
-    contentfulPreviewSpaceId: Joi.string().description(
-      'Space ID from Contentful Dashboard settings'
+    nacelleSpaceId: Joi.string().description(
+      'Space ID from the Nacelle Dashboard'
+    ),
+    nacelleGraphqlToken: Joi.string().description(
+      'GraphQL Token from the Nacelle Dashboard'
+    ),
+    nacelleClient: Joi.optional().description(
+      'Instance of the `@nacelle/client-js-sdk` which you want to use to fetch your nacelle data'
     ),
```

Comment:
Two things:

1. It looks like `contentfulPreviewSpaceId` is being removed, but should be there.
2. It looks like we were missing `nacelleEndpoint` from the `Joi` validation - this was an existing issue, but seems like a good time to fix it.

```suggestion
    nacelleSpaceId: Joi.string().description(
      'Space ID from the Nacelle Dashboard'
    ),
    nacelleGraphqlToken: Joi.string().description(
      'GraphQL Token from the Nacelle Dashboard'
    ),
    nacelleEndpoint: Joi.string().description(
      'Storefront Endpoint from the Nacelle Dashboard'
    ),
    nacelleClient: Joi.optional().description(
      'Instance of the `@nacelle/client-js-sdk` which you want to use to fetch your nacelle data'
    ),
    contentfulPreviewSpaceId: Joi.string().description(
      'Space ID from Contentful Dashboard settings'
    ),
```

Reply:
Good catch! Should be fixed